cg/server/admin.py

```diff
@@ -429,6 +430,71 @@ def view_sample_link(unused1, unused2, model, unused3):
             else ""
         )
 
+    @action(
+        "cancel_samples",
+        "Cancel samples",
+        "Are you sure you want to cancel the selected samples?",
+    )
+    def cancel_samples(self, entry_ids: List[str]) -> None:
+        """
+        Action for cancelling samples:
+            - Comments each sample being cancelled with date and user.
+            - Deletes any relationship between the cancelled samples and cases.
+            - Deletes any cases that only contain samples being cancelled.
+        """
+        all_associated_case_ids: set = set()
+
+        for entry_id in entry_ids:
+            sample: Sample = db.get_sample_by_id(entry_id=entry_id)
```

Comment:
Is not entry_id a str according to ` entry_ids: List[str]`, but the function expects an int? Or I might be missing something

Reply:
Yes, it is confusing. I should change it to [int] instead.